bin/client/src/kona.rs

```diff
@@ -57,12 +57,14 @@ fn main() -> Result<()> {
         .await?;
         let L2AttributesWithParent { attributes, .. } = driver.produce_disputed_payload().await?;
 
-        let precompile_overrides =
-            FPVMPrecompileOverride::<OracleL2ChainProvider, TrieDBHintWriter>::default();
+        let precompile_overrides = FPVMPrecompileOverride::<
+            OracleL2ChainProvider<CachingOracle>,
+            OracleL2ChainProvider<CachingOracle>,
+        >::default();
```

Comment:
@clabby I understand the logic for keeping a separate Fetcher and Hinter in the library crates, as we want to keep flexibility for future users to implement these separately.

But if we are going to have the L2 Provider fulfill both of these roles in `kona-client`, do you think it makes sense to refactor `FPVMPrecompileOverride` to just be generic over a single type that implements both traits?

Reply:
Since this simplifies the ux and isn't clear who will need the split logic, I support doing this refactor in a follow-on pr. The FPVM Precompile Override can always be made even more abstract by encapsulating it's internal logic in a trait, so not too worried about increasing the abstractions here.